android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java

```diff
@@ -525,11 +608,22 @@ public void propertyChanged(String key, Object oldValue, Object newValue, KrollP
 		} else if (TiC.PROPERTY_SCROLLING_ENABLED.equals(key)) {
 			setScrollingEnabled(newValue);
 		} else if (TiC.PROPERTY_REFRESH_CONTROL.equals(key)) {
-			if (newValue instanceof RefreshControlProxy) {
-				Log.w(TAG, REFRESH_CONTROL_NOT_SUPPORTED_MESSAGE);
+			View nativeView = getNativeView();
+			if (newValue == null) {
```

Comment:
This could be simplified to:
```Java
if (nativeView instanceof TiSwipeRefreshLayout) {
	RefreshControlProxy.unassignFrom((TiSwipeRefreshLayout) nativeView);

	if (newValue instanceof RefreshControlProxy) {
		((RefreshControlProxy) newValue).assignTo((TiSwipeRefreshLayout) nativeView);
	}
} else {
	Log.e(TAG, "Invalid value assigned to property '" + key + "'. Must be of type 'RefreshControl'.");
}
```